### PR TITLE
Automatic deployment integration

### PR DESCRIPTION
**Pull request by AronNovak**
## Status
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -130,3 +130,18 @@ To Deploy to a Pantheon environment (e.g. TEST or LIVE) you can use
 
     # Deploy to LIVE.
     ddev robo deploy:pantheon-sync live
+
+## Automatic deployment to Pantheon
+
+In order to deploy upon every merge automatically by Travis, you shall:
+
+1. Install https://github.com/travis-ci/travis.rb
```
Comment (amitaibu) [inline README.md:138]: would be nicer not to have to install anything. In fact, I think we could have a robo command to take care of all the steps (getting as argument the terminus token which was created by manually), so it all happens inside the ddev container.

Comment (AronNovak) [reply README.md:138]: That's right, soon I will work on a bit more to automate that.

**Comment by AronNovak** [comment]
image

**Comment by AronNovak** [comment]
@amitaibu I restructured this a bit and added the error handling too.

**Comment by amitaibu** [comment]
:+1: Looks really nice

File: .travis.yml [hunk 2]
```diff
@@ -34,5 +40,13 @@ jobs:
     - stage: Test
       name: "Backend tests: Functional tests"
       script:
-        - "$TRAVIS_BUILD_DIR/ci-scripts/install_server.sh"
+        - "$TRAVIS_BUILD_DIR/ci-scripts/install_ddev.sh"
+        - "$TRAVIS_BUILD_DIR/ci-scripts/install_drupal.sh"
         - "$TRAVIS_BUILD_DIR/ci-scripts/test_phpunit.sh"
+    - stage: Deploy
+      name: "Deploy to Pantheon"
+      if: branch = editmetomakeitwork AND type = push
```
Comment (amitaibu) [inline .travis.yml:48]:
```suggestion
      if: branch = master AND type = push
```

File: README.md [hunk 3]
```diff
@@ -130,3 +130,16 @@ To Deploy to a Pantheon environment (e.g. TEST or LIVE) you can use
 
     # Deploy to LIVE.
     ddev robo deploy:pantheon-sync live
+
+## Automatic deployment to Pantheon
+
+In order to deploy upon every merge automatically by Travis, you shall:
+
+1. Get a Pantheon machine token (on a dummy new Pantheon user ideally): https://pantheon.io/docs/machine-tokens
```
Comment (amitaibu) [inline README.md:138]:
```suggestion
1. Get a Pantheon machine token (or a dummy new Pantheon user ideally): https://pantheon.io/docs/machine-tokens
```

Why dummy user is better?

File: README.md [hunk 4]
```diff
@@ -130,3 +130,16 @@ To Deploy to a Pantheon environment (e.g. TEST or LIVE) you can use
...
+1. `ddev deploy:config-autodeploy [your new token]`
+1. `git commit -m "Deployment secrets"`
+1. Add the public key in `travis-key.pub` to the newly created dummy Pantheon user: https://pantheon.io/docs/ssh-keys
+1. Edit `.travis.yml`, replace `replaceme.drush.in`, `ssh://something@replaceme.drush.in:repository` and `editmetomakeitwork` with real values. `terminus connection:info yoursite.dev` gives you the information for `GIT_HOST` and `PANTHEON_GIT_URL`. `editmetomakeitwork` shall be usually `master`, meaning that when you merge something to `master` on GitHub, that triggers a deploy.
```
Comment (amitaibu) [inline README.md:142]: lest have this as part of the robo command. So it really takes care of everything it can do for us

File: .travis.yml (hunk 2 above)
Comment (AronNovak) [reply .travis.yml:48]: But we do not want that by default, as it would make Travis fail on a clone of this repo immediately.

Comment (amitaibu) [reply .travis.yml:48]: Got it. Maybe change then to `{{deploy-branch}}`, so Robo command would know to replace it?

File: README.md (hunk 3 above)
Comment (AronNovak) [reply README.md:138]: As the token is not per-project, we do not want to compromise all the sites of a developer when a Travis environment is compromised for one project.

Comment (AronNovak) [reply README.md:138]: The dummy user should be part of only one project in the end.

**Comment by AronNovak** [comment]
image

**Comment by AronNovak** [comment]
In https://github.com/Gizra/drupal-starter/pull/25, I actually try this command out.

**Comment by AronNovak** [comment]
@amitaibu This is ready for another review.

**Comment by AronNovak** [comment]
image
image

**Pull request merged**